src/Widgets/TerminalWidget.vala

```diff
@@ -519,46 +519,41 @@ namespace Terminal {
 
         protected override void paste_clipboard () {
             clipboard.request_text ((clipboard, text) => {
-                if (text == null) {
-                    return;
-                }
-
-                if (!text.validate ()) {
-                    warning ("Dropping invalid UTF-8 paste");
-                    return;
-                }
+                validated_feed (text);
+            });
+        }
 
+        // Check pasted and dropped text before feeding to child;
+        private void validated_feed (string? text) {
+            if (text != null && text.validate ()) {
+                var stripped_text = text.strip ();
                 unowned var toplevel = (MainWindow) get_toplevel ();
+                if (!toplevel.unsafe_ignored &&
+                    Application.settings.get_boolean ("unsafe-paste-alert")) {
 
-                if (!toplevel.unsafe_ignored && Application.settings.get_boolean ("unsafe-paste-alert")) {
                     string? warn_text = null;
-                    text._strip ();
```

Comment:
I think we should call `var stripped_text = text.strip ();` here instead of L529 for lesser scope.

Reply:
Removed var altogether